### ISSUE TITLE
Understand the reason behind the warp block?

### ISSUE THREAD

**Issue by stereomatchingkiss** [sangyun884/HR-VITON#22]
In the file networks.py, line 133~135

```
flow_norm = torch.cat([flow[:, :, :, 0:1] / ((iW/2 - 1.0) / 2.0), flow[:, :, :, 1:2] / ((iH/2 - 1.0) / 2.0)], 3)
warped_T1 = F.grid_sample(T1, flow_norm + grid, padding_mode='border')

```                
Why do you need to normalize the value of the flow?
Why do you sum the flow_norm with grid? 
If we do not perform norm and/or sum the flow_norm with the grid, what are the impact to the final results?
Any paper mention about this warp block? I read your paper but it do not give much explanation about this warp block(The W symbol in the Fig10).

Thanks

**Comment by Gzzgz** [comment]
I have the same problem.
Do you know the code " flow[:, :, :, 0:1] / ((iW/2 - 1.0) / 2.0 " 's mean?

**Comment by stereomatchingkiss** [comment]
> 

Same here, don't know why they norm the flow like that

**Comment by stereomatchingkiss** [comment]
> I have the same problem. Do you know the code " flow[:, :, :, 0:1] / ((iW/2 - 1.0) / 2.0 " 's mean?

[A good explanation](https://discuss.pytorch.org/t/warp-video-frame-from-optical-flow/6013/4)
